typings/index.ts

```diff
@@ -398,13 +398,38 @@ client.on('ready', async () => {
   });
 });
 
+// This is to check that stuff is the right type
+declare const assertIsMember: (m: Promise<GuildMember>) => void;
+
 client.on('guildCreate', g => {
   const channel = g.channels.cache.random();
   if (!channel) return;
 
   channel.setName('foo').then(updatedChannel => {
     console.log(`New channel name: ${updatedChannel.name}`);
   });
+
+  // @ts-expect-error
+  assertIsMember(g.members.add(testUserId));
+
+  // @ts-expect-error
+  assertIsMember(g.members.add(testUserId, {}));
+
+  // @ts-expect-error
+  assertIsMember(g.members.add(testUserId, { accessToken: 'totallyRealAccessToken', roles: [g.roles.cache] }));
+
+  // @ts-expect-error
+  assertIsMember(g.members.add(testUserId, { accessToken: 'totallyRealAccessToken', fetchWhenExisting: false }));
+
+  assertIsMember(
+    g.members.add(testUserId, {
+      accessToken: 'totallyRealAccessToken',
+      mute: true,
+      deaf: false,
+      roles: [g.roles.cache.first()!],
+      force: true,
```

Comment:
Probably want to test when it's set to `true`, so we test all three cases (`undefined`, `false`, and `true`):
```suggestion
      roles: [g.roles.cache.first()!],
      fetchWhenExisting: true,
      force: true,
```